Dockerfile

```diff
@@ -1,5 +1,6 @@
 # build stage
 FROM --platform=$BUILDPLATFORM golang:1.15-alpine AS build-env
+ARG TARGETPLATFORM
```

Comment:
So the docker build-push action uses the values we define for [platforms](https://github.com/aws/aws-xray-daemon/blob/3ec6dfb60e18eafdec9dbc593c5fbf7282149e4c/.github/workflows/release-build-publish.yml#L79) as this argument to `docker build`? That's a little bit of magic

Reply:
Yes docker provides `TARGETPLATFORM` (this will be both `linux/amd64 and linux/arm64` whatever we provide here - https://github.com/aws/aws-xray-daemon/blob/3ec6dfb60e18eafdec9dbc593c5fbf7282149e4c/.github/workflows/release-build-publish.yml#L79 ) and `BUILDPLATFORM` (this value will be where on which platform we are running this docker file) - More explanation here (https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images)